src/Lecture1.hs

```diff
@@ -104,7 +118,13 @@ and finds a sum of the numbers inside this string.
 
 The string contains only spaces and/or numbers.
 -}
-strSum str = error "TODO"
+strSum :: [Char] -> Int
+strSum str = go 0 (words str)
+    where
+        go :: Int -> [[Char]] -> Int
+        go result list
+            | null list = result
+            | otherwise = go (result + read (head list) :: Int) (tail list)
```

Comment:
There's already a standard function `sum` for finding the sum of all elements in the list ➕ 

As you specify the top-level type signature for this function, the compiler would be able to figure out what type `read` function has in this case due to type inference, so you can skip specifying it in the body:

```suggestion
            | otherwise = go (result + read (head list)) (tail list)
```

So, if you use standard functions `sum` and `map`, you'll be able to implement `strSum` as a nice one-liner